src/components/Filters/types.ts

```diff
@@ -1,22 +1,24 @@
-import { Key } from "react";
-import { MultiSelectFieldProps, SelectFieldProps } from "src/inputs";
+import { MultiSelectFieldProps, SelectFieldProps, Value } from "src/inputs";
 
-export type FilterDefs<F> = { [K in keyof F]: FilterDef<F[K]> };
+export type FilterDefs<F> = {
+  [K in keyof F]: FilterDef<Exclude<F[K], null | undefined>>;
+};
 
-export type SingleFilterProps<O, V extends Key> = Omit<SelectFieldProps<O, V>, "value" | "onSelect">;
-export type MultiFilterProps<O, V extends Key> = Omit<MultiSelectFieldProps<O, V>, "values" | "onSelect">;
+export type SingleFilterProps<O, V extends Value> = Omit<SelectFieldProps<O, V>, "value" | "onSelect">;
+export type MultiFilterProps<O, V extends Value> = Omit<MultiSelectFieldProps<O, V>, "values" | "onSelect">;
 export type BooleanOption = [boolean | undefined, string];
 
 // What is V?
 // - V might be `string[]` and could be used for a multiselect that getOptionValue returned strings
 // - V might be `number[]` and could be used for a multiselect that getOptionValue returned numbers
 // - V might be `boolean` and could be used for ...boolFilter...
-export type FilterDef<V> = V extends Array<infer U>
-  ? U extends Key
-    ? { kind: "multi" } & MultiFilterProps<any, U>
-    : never
-  : V extends boolean | undefined
+
+// All of the extra brackets are to avoid `Stage` turning into `Stage.One` | `Stage.Two`:
+// https://stackoverflow.com/questions/53996797/typescript-conditional-type-array-of-union-type-distribution
```

Comment:
This `FilterDef` change is the core fix; specifically moving from `V extends Key` to `[V] extends [Key]` (actually `[Value]` here as a separate/related clean up).